### DISCUSSION TITLE
I need 2 column relationship in one relation

### DISCUSSION THREAD

**Discussion by fvthakor** [sequelize/sequelize#15939] [Help & Questions]
here is my sql table 

CREATE TABLE `Event_User_Requests_Song` (
  `Event_User_Request_Song_ID` bigint NOT NULL AUTO_INCREMENT,
  `Song_ID` bigint NOT NULL,
  `User_Joins_Event_ID` bigint NOT NULL,
  `Event_ID_Index` bigint NOT NULL,
  `Song_Approval_Status` varchar(10) NOT NULL,
  PRIMARY KEY (`Event_User_Request_Song_ID`),
  UNIQUE KEY `Unique_Song_Request` (`Song_ID`,`Event_ID_Index`),
  KEY `fk_Song_has_Users_Joins_Events_Users_Joins_Events1_idx` (`User_Joins_Event_ID`,`Event_ID_Index`),
  KEY `fk_Song_has_Users_Joins_Events_Song1_idx` (`Song_ID`),
  CONSTRAINT `fk_Song_has_Users_Joins_Events_Song1` FOREIGN KEY (`Song_ID`) REFERENCES `Song` (`Song_ID`),
  CONSTRAINT `fk_Song_has_Users_Joins_Events_Users_Joins_Events1` FOREIGN KEY (`User_Joins_Event_ID`, `Event_ID_Index`) REFERENCES `Users_Joins_Events` (`User_Joins_Event_ID`, `Event_ID_Index`)
) ENGINE=InnoDB AUTO_INCREMENT=121 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;

CREATE TABLE `Users_Joins_Events` (
  `User_Joins_Event_ID` bigint NOT NULL AUTO_INCREMENT,
  `Event_ID_Index` bigint NOT NULL,
  `User_ID` int NOT NULL,
  `Is_Event_Manager` tinyint(1) DEFAULT NULL,
  `Is_Banned_From_Event` tinyint(1) DEFAULT NULL,
  PRIMARY KEY (`User_Joins_Event_ID`,`Event_ID_Index`,`User_ID`),
  UNIQUE KEY `User_Joins_Event_ID_UNIQUE` (`User_Joins_Event_ID`),
  KEY `fk_Event_has_users_Event1_idx` (`Event_ID_Index`),
  KEY `fk_Event_has_users_users1_idx` (`User_ID`),
  CONSTRAINT `fk_Event_has_users_Event1` FOREIGN KEY (`Event_ID_Index`) REFERENCES `Event` (`Event_ID_Index`),
  CONSTRAINT `fk_Event_has_users_users1` FOREIGN KEY (`User_ID`) REFERENCES `users` (`user_id_index`) ON DELETE RESTRICT ON UPDATE RESTRICT
) ENGINE=InnoDB AUTO_INCREMENT=25 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_general_ci;

I need relationship like that 
![image](https://user-images.githubusercontent.com/42373267/232663643-56d2ae09-de08-4723-8673-5be77fd9d253.png)

can you please help me

**Comment by fvthakor** [comment]
also I tried that 

db.EventUserRequestsSong.belongsTo(db.UsersJoinsEvent, {
    foreignKey: ['User_Joins_Event_ID', 'Event_ID_Index']
});

db.UsersJoinsEvent.hasMany(db.EventUserRequestsSong, {
    foreignKey: ['User_Joins_Event_ID', 'Event_ID_Index'],
    onDelete: 'RESTRICT',
    onUpdate: 'RESTRICT'
});
 it's create new column in `Event_User_Requests_Song`
![image](https://user-images.githubusercontent.com/42373267/232664007-a4c4eefb-0938-418a-b894-010db089973d.png)

**Comment by laurelgr** [comment]
From what I read on [this issue](https://github.com/sequelize/sequelize/issues/311) it is not yet possible, but might be under consideration.

Possible workarounds on [stackoverflow](https://stackoverflow.com/questions/29015156/sequelize-composite-foreign-key)

For formatting: if you want to write multiple lines of code (like a table definition), you can put it within triple ``` instead of single ` 😉 
```
db.EventUserRequestsSong.belongsTo(db.UsersJoinsEvent, {
foreignKey: ['User_Joins_Event_ID', 'Event_ID_Index']
});
```

**Comment by fvthakor** [comment]
Okay Thanks
